core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -162,8 +162,8 @@ private void initializeCatalogTables() {
             DatabaseMetaData dbMeta = conn.getMetaData();
             ResultSet tableExists =
                 dbMeta.getTables(
-                    null /* catalog name */,
-                    null /* schemaPattern */,
+                    conn.getCatalog() /* catalog name */,
+                    conn.getSchema() /* schemaPattern */,
```

Comment:
The 2nd argument needs an escape. Same for `NAMESPACE_PROPERTIES_TABLE_NAME`. 
Please take a look at #11863

Reply:
yes, the schema needs an escape.